### PR TITLE
doc: fix invalid link 03-client-state.md

### PR DESCRIPTION
**Pull request by osrm**
## Description

 

From the documentation at https://ibc.cosmos.network/v8/ibc/light-clients/localhost/client-state#client-updates

"[See BeginBlocker in abci.go.](https://github.com/cosmos/ibc-go/blob/09-localhost/modules/core/02-client/abci.go#L12)" link  is invalid.

The correct path for that link is 
"https://github.com/cosmos/ibc-go/blob/v8.5.0/modules/core/02-client/abci.go#L12". 
(Since this is a versionised docs, a link to the same version, 8.5.0, is appropriate.)

## Changes

https://github.com/cosmos/ibc-go/blob/6c415051b69c3d01706fc1a1957293cb819e955a/docs/versioned_docs/version-v8.5.x/03-light-clients/02-localhost/03-client-state.md?plain=1#L36

Also briefly mentioned in commit, 

"https://github.com/cosmos/ibc-go/blob/09-localhost/modules/core/02-client/abci.go#L12" -> "https://github.com/cosmos/ibc-go/blob/v8.5.0/modules/core/02-client/abci.go#L12"

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [x] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/versioned_docs/version-v8.5.x/03-light-clients/02-localhost/03-client-state.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ It is possible to disable the localhost client by removing the `09-localhost` en
 
 The latest height is updated periodically through the ABCI [`BeginBlock`](https://docs.cosmos.network/v0.47/building-modules/beginblock-endblock) interface of the 02-client submodule in core IBC.
 
-[See `BeginBlocker` in abci.go.](https://github.com/cosmos/ibc-go/blob/09-localhost/modules/core/02-client/abci.go#L12)
+[See `BeginBlocker` in abci.go.](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/abci.go#L12)
```
Comment (gjermundgaraba) [inline docs/versioned_docs/version-v8.5.x/03-light-clients/02-localhost/03-client-state.md:36]: I don't think this link is quite correct either. It is not dead, but it is also not going to where it should :)

**Review by gjermundgaraba — CHANGES_REQUESTED** [review]

File: docs/versioned_docs/version-v8.5.x/03-light-clients/02-localhost/03-client-state.md (hunk 1 above)
Comment (osrm) [reply docs/versioned_docs/version-v8.5.x/03-light-clients/02-localhost/03-client-state.md]: Agree with that, I just modified that to L28. Could you please check it?

'The latest height is updated periodically via the ABCI BeginBlock interface on the 02 client submodule of the core IBC.

See BeginBlocker at abci.go.'

I believe this corresponds to 

https://github.com/cosmos/ibc-go/blob/e10bbbc47fb70e138a3369902d18586b3da95800/modules/core/02-client/abci.go#L28

File: docs/versioned_docs/version-v8.5.x/03-light-clients/02-localhost/03-client-state.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ It is possible to disable the localhost client by removing the `09-localhost` en
 
 The latest height is updated periodically through the ABCI [`BeginBlock`](https://docs.cosmos.network/v0.47/building-modules/beginblock-endblock) interface of the 02-client submodule in core IBC.
 
-[See `BeginBlocker` in abci.go.](https://github.com/cosmos/ibc-go/blob/09-localhost/modules/core/02-client/abci.go#L12)
+[See `BeginBlocker` in abci.go.](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/abci.go#L28)
```
Comment (gjermundgaraba) [inline docs/versioned_docs/version-v8.5.x/03-light-clients/02-localhost/03-client-state.md:36]: Given that this is in the versioned docs, I would prefer to change the link to use the code from the same branch: https://github.com/cosmos/ibc-go/tree/release/v8.5.x

Please also take a look and make sure the link is correct in the other versioned docs.

**Review by gjermundgaraba — CHANGES_REQUESTED** [review]

File: docs/versioned_docs/version-v8.5.x/03-light-clients/02-localhost/03-client-state.md (hunk 2 above)
Comment (osrm) [reply docs/versioned_docs/version-v8.5.x/03-light-clients/02-localhost/03-client-state.md]: I followed your advice and applied the changes to the versioned docs. 

The link https://github.com/cosmos/ibc-go/blob/3077be12f11e9859609028d2b27971a8c6a8fab1/modules/core/02-client/abci.go#L12 is right. I missed.

As a result, I updated the link path and modified pr.

Additionally, I reviewed other versions of the versioned documents. The path for version 7.8.x is correct, and no documentation exists for versions other than 8.5.x.

Therefore, I concluded that only the 8.5.x version needed fixing.

Thank you for your advice. If you notice any further issues or mistakes in my work, please let me know.

**Review by gjermundgaraba — APPROVED** [review]
Excellent! Thank you :)

**Pull request merged**